File: 523-continuous-subarray-sum/523. Continuous Subarray Sum.java
import java.util.HashMap;

class Solution {
    public boolean checkSubarraySum(int[] nums, int k) {
        int sum = 0;
        HashMap<Integer , Integer> hm  = new HashMap<Integer, Integer>();
        hm.put(0,-1);
        for(int i = 0; i < nums.length; i++){
            sum+=nums[i];
            int rem = sum %k;
            if(hm.containsKey(rem)){
                int ind = hm.get(rem);
                if(i - ind >=2){
                    return true;
                }
            }
            else{
                hm.put(rem,i);
            }
        }
        return false;
    }
}

File: 523-continuous-subarray-sum/523. Continuous Subarray Sum.cpp
class Solution {
public:
    bool checkSubarraySum(vector<int>& nums, int k) {
        // ... (implementation of the checkSubarraySum function)
        if(nums.size()<=1){
            return 0;
        }
        unordered_map<int,int>remindercheck ;
        remindercheck[0]=-1;
        long long int sum =0;
        for(int i =0;i<nums.size();i++){
            sum +=nums[i];
            int rem = sum%k;
            if(remindercheck.count(rem)){
                int previndx = remindercheck[rem];
                if(i-previndx>=2){
                    return true;
                }
            }            
            else{
                remindercheck[rem] =i;
            }

        }
        return false;

        

    }
};